### DISCUSSION TITLE
How to enable pawn rotation?

### DISCUSSION THREAD

**Discussion by Jus2Cat** [Sixze/ALS-Refactored#122] [Q&A]
Hello! I have my own settings for FP. How Can I disable rotation at angle and turn on the absolute rotation? If I'm trying to disable update Grounded Rotation (tick function). After that pawn rotation is not working at all. I tried increase rotation rate and set bUseControllerRotationYaw to true. It has no effect.  I can't make him rotate. 

ps. It can be good make something like if (RotationMode != Aiming && RotationMode != LookingDirection && RotationMode != VelocityDirection) { make rotation possible } or one more checkbox in the settings

**Comment by Jus2Cat** [comment]
Ok. I found solution. I didn't know about Face Rotation() function overriding. Parent logic was skipped so rotation was not updated with ControlRotation
![image](https://user-images.githubusercontent.com/48173145/180155855-0e317791-1ca8-4a5b-b46f-4275145d729e.png)

/** Updates Pawn's rotation to the given rotation, assumed to be the Controller's ControlRotation. Respects the bUseControllerRotation* settings. */

**Comment by Sixze** [comment]
You can set `ViewMode` to `FirstPerson` and inside `AAlsCharacter::RefreshGroundedMovingAimingRotation()` change `ViewRelativeYawAngleThreshold` to `0.0f`.

**Reply by Jus2Cat** [reply]
Thank you for your answer! But this solution has a jitters and a little rotation delay. And also I can use my own view mode (AimDownSight)

**Reply by Jus2Cat** [reply]
> You can set `ViewMode` to `FirstPerson` and inside `AAlsCharacter::RefreshGroundedMovingAimingRotation()` change `ViewRelativeYawAngleThreshold` to `0.0f`.

With FP view mode and ViewRelativeYawAngleThreshold = 0.0f.
![Sixze](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif)

With bUseControllerRotationYaw = true and FaceRotation()
![Me](https://user-images.githubusercontent.com/48173145/180202519-07dfd8cc-bb54-407e-803a-4a4ff7357d49.gif)

**Reply by falbellaihi1** [reply]
Nice, I went another route just made the upper body rotate instead of head in ALS_View has its problems tho

https://user-images.githubusercontent.com/9819280/180295700-26a40f4b-e052-4abd-a392-da627aef2985.mp4



https://user-images.githubusercontent.com/9819280/180300951-22dc9fec-0bcc-4d9c-823e-4f24a5d08e92.mp4

**Reply by catarrestor** [reply]
> > You can set `ViewMode` to `FirstPerson` and inside `AAlsCharacter::RefreshGroundedMovingAimingRotation()` change `ViewRelativeYawAngleThreshold` to `0.0f`.
> 
> With FP view mode and ViewRelativeYawAngleThreshold = 0.0f. ![Sixze](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif) [ ![Sixze](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif) ](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif) [ ](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif)
> 
> With bUseControllerRotationYaw = true and FaceRotation() ![Me](https://user-images.githubusercontent.com/48173145/180202519-07dfd8cc-bb54-407e-803a-4a4ff7357d49.gif)



> > You can set `ViewMode` to `FirstPerson` and inside `AAlsCharacter::RefreshGroundedMovingAimingRotation()` change `ViewRelativeYawAngleThreshold` to `0.0f`.
> 
> With FP view mode and ViewRelativeYawAngleThreshold = 0.0f. ![Sixze](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif) [ ![Sixze](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif) ](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif) [ ](https://user-images.githubusercontent.com/48173145/180202530-b33f0918-aa1a-4c9c-8981-d8a2af7eea05.gif)
> 
> With bUseControllerRotationYaw = true and FaceRotation() ![Me](https://user-images.githubusercontent.com/48173145/180202519-07dfd8cc-bb54-407e-803a-4a4ff7357d49.gif)

Hi everyone, sorry little of topic question how do I implement to play montages in this Al’s, I followed a couple tutorials from YouTube but can’t implement character firing animation, the only difference I saw It doesn’t have layering override slots, I’m not sure if that is my problem, because in main animation blueprint there is a comment  stating to play montages there and transfer the output pose to linked animation blueprints. If someone can post a pictures of blueprints or can explain how to implement the firing animation on character would be awesome I think I been banging my head on this for 2 days now. Thanks

**Comment by Sixze** [comment]
As an alternative to all the methods above, you can simply replace `View.Rotation` with `View.NetworkSmoothing.Rotation` inside `UAlsAnimationInstance::RefreshViewGameThread()`.

**Reply by hskldr** [reply]
Does this still work?
